dvc/repo/experiments/executor/base.py

```diff
@@ -450,9 +452,32 @@ def _repro_args(cls, dvc):
             kwargs = {}
         return args, kwargs
 
+    @staticmethod
+    def _auto_push(git_remote: str, dvc: "Repo", scm: "Git"):
+        from dvc.repo.experiments.push import push
+
+        if git_remote == scm.root_dir:
+            logger.warning(
+                f"{DVC_EXP_AUTO_PUSH} {git_remote} is the running "
+                "repository auto push will not work"
+            )
+            return
```

Comment:
Have you tested this behavior (and did the push actually fail)?

I would not expect this to error out, pushing to yourself is possible with CLI git, and just does nothing, since the refs will always be up to date.

A warning might be appropriate since the user was probably not trying to push to self, but I don't think this should be an error condition

Reply:
Actually now that I'm thinking about it, one situation where this "push to self" behavior may be useful is if the user wants to ensure that DVC cache is always pushed for their experiments, but does not care about pushing the individual exp commits to a git remote.